diff -ur mdadm-2.2.org/Assemble.c mdadm-2.2/Assemble.c
--- mdadm-2.2.org/Assemble.c	2005-12-05 06:56:20.000000000 +0100
+++ mdadm-2.2/Assemble.c	2006-01-05 20:36:50.030283750 +0100
@@ -219,7 +219,7 @@
 		}
 		if (dfd >= 0) close(dfd);
 
-		if (ident->uuid_set && (!update && strcmp(update, "uuid")!= 0) &&
+		if (ident->uuid_set && (update && strcmp(update, "uuid")!= 0) &&
 		    (!super || same_uuid(info.uuid, ident->uuid, tst->ss->swapuuid)==0)) {
 			if ((inargv && verbose >= 0) || verbose > 0)
 				fprintf(stderr, Name ": %s has wrong uuid.\n",
