--- mdadm-1.6.0/Assemble.c.orig	Fri Jun  4 08:18:28 2004
+++ mdadm-1.6.0/Assemble.c	Tue Jul 13 10:33:52 2004
@@ -253,14 +253,14 @@
 		 */
 
 		if (!havesuper) {
-			fprintf(stderr, Name ": %s has no superblock - assembly aborted\n",
+			fprintf(stderr, Name ": %s has no superblock - device ignored\n",
 				devname);
-			return 1;
+			continue;
 		}
 		if (compare_super(&first_super, &super)) {
-			fprintf(stderr, Name ": superblock on %s doesn't match others - assembly aborted\n",
+			fprintf(stderr, Name ": superblock on %s doesn't match others - device ignored\n",
 				devname);
-			return 1;
+			continue;
 		}
 
 
