diff -urN mdadm-1.0.1.org/Monitor.c mdadm-1.0.1/Monitor.c
--- mdadm-1.0.1.org/Monitor.c	Sat Sep  7 21:10:30 2002
+++ mdadm-1.0.1/Monitor.c	Sat Sep  7 21:11:01 2002
@@ -271,7 +271,7 @@
 		if (scan) {
 			struct mdstat_ent *mse;
 			for (mse=mdstat; mse; mse=mse->next) 
-				if (mse->devnum > 0) {
+				if (mse->devnum >= 0) {
 					struct state *st = malloc(sizeof *st);
 					if (st == NULL)
 						continue;
