--- mdadm-2.6.7.1/super0.c~	2008-11-05 08:39:03.768811554 +0100
+++ mdadm-2.6.7.1/super0.c	2008-11-05 08:46:21.492309369 +0100
@@ -847,6 +847,9 @@
 	st->minor_version = 90;
 	st->max_devs = MD_SB_DISKS;
 	st->sb = NULL;
+	/* Eliminate pointless leading 0 from some versions of mdadm -D */
+	if (strncmp(arg, "00.", 3) == 0)
+		arg++;
 	if (strcmp(arg, "0") == 0 ||
 	    strcmp(arg, "0.90") == 0 ||
 	    strcmp(arg, "default") == 0 ||
