--- mdadm-2.5.1/Assemble.c	2006-06-02 08:54:03.000000000 +0200
+++ mdadm-2.5.1.new/Assemble.c	2006-06-17 11:36:33.161117670 +0200
@@ -313,10 +313,9 @@
 		 */
 
 		if (!super) {
-			fprintf(stderr, Name ": %s has no superblock - assembly aborted\n",
+			fprintf(stderr, Name ": %s has no superblock - device ignored\n",
 				devname);
-			free(first_super);
-			return 1;
+			continue;
 		}
 
 		if (st == NULL)
@@ -354,11 +353,9 @@
 					}
 				}
 			}
-			fprintf(stderr, Name ": superblock on %s doesn't match others - assembly aborted\n",
+			fprintf(stderr, Name ": superblock on %s doesn't match others - device ignored\n",
 				devname);
-			free(super);
-			free(first_super);
-			return 1;
+			continue;
 		}
 
 		tmpdev->used = 1;
