--- mdadm-3.0/mdassemble.c.orig	2009-06-02 07:48:29.000000000 +0200
+++ mdadm-3.0/mdassemble.c	2009-09-17 16:30:34.471798993 +0200
@@ -106,7 +106,7 @@
 				close(mdfd);
 			rv |= Assemble(array_list->st, array_list->devname,
 				       array_list, NULL, NULL,
-				       readonly, runstop, NULL, NULL, 0,
+				       readonly, runstop, NULL, "<unknown>", 0,
 				       verbose, force);
 		}
 	return rv;
