diff -urN mdadm-1.4.0.org/mdadm.h mdadm-1.4.0/mdadm.h
--- mdadm-1.4.0.org/mdadm.h	2004-01-21 14:45:56.918173309 +0100
+++ mdadm-1.4.0/mdadm.h	2004-01-21 15:59:55.391466540 +0100
@@ -49,7 +49,7 @@
 #define	MD_MAJOR 9
 
 #ifndef BLKGETSIZE64
-#define BLKGETSIZE64 _IOR(0x12,114,sizeof(__u64)) /* return device size in bytes (u64 *arg) */
+#define BLKGETSIZE64 _IOR(0x12,114,size_t) /* return device size in bytes (u64 *arg) */
 #endif
 
 
