diff -urN mdadm-2.0.org/Assemble.c mdadm-2.0/Assemble.c
--- mdadm-2.0.org/Assemble.c	2005-08-15 08:31:57.000000000 +0200
+++ mdadm-2.0/Assemble.c	2005-09-04 00:25:53.575606024 +0200
@@ -262,18 +262,15 @@
 		 */
 
 		if (!super) {
-			fprintf(stderr, Name ": %s has no superblock - assembly aborted\n",
+			fprintf(stderr, Name ": %s has no superblock - device ignored\n",
 				devname);
-			free(first_super);
-			return 1;
+			continue;
 		}
 		st = tst; /* commit to this format, if haven't already */
 		if (st->ss->compare_super(&first_super, super)) {
-			fprintf(stderr, Name ": superblock on %s doesn't match others - assembly aborted\n",
+			fprintf(stderr, Name ": superblock on %s doesn't match others - device ignored\n",
 				devname);
-			free(super);
-			free(first_super);
-			return 1;
+			continue;
 		}
 
 		/* looks like a good enough match to update the super block if needed */
